Lib/typing.py

```diff
@@ -1400,10 +1394,7 @@ def _allow_reckless_class_checks(depth=3):
     The abc and functools modules indiscriminately call isinstance() and
     issubclass() on the whole MRO of a user class, which may contain protocols.
     """
-    try:
-        return sys._getframe(depth).f_globals['__name__'] in ['abc', 'functools']
-    except (AttributeError, ValueError):  # For platforms without _getframe().
-        return True
+    return _callee(depth) in {'abc', 'functools', None}
```

Comment:
The meaning here seems slightly different from the old code. If `__name__ = None`, the old code would return `False`, but the new code will return `True` right?

I recommend
```suggestion
    return _callee(depth, unsupported=object()) in {'abc', 'functools', object()}
```
or some equivalent sentinel.

Reply:
Oh actually I thought of a small micro-optimization. If you do
```
def _allow_reckless_class_checks(depth=3, sentinel=object()):
    ...
    return _callee(depth, unsupported=sentinel) in {'abc', 'functools', sentinel}
```
we can avoid creating the `object()` every time the function is called. This may help runtime protocols.

Reply:
Thanks, that makes sense.